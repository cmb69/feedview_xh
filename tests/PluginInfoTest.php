<?php

/**
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Feedview_XH.
 *
 * Feedview_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Feedview_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Feedview_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Feedview;

use ApprovalTests\Approvals;
use Feedview\Infra\FakeSystemChecker;
use Feedview\Infra\View;
use PHPUnit\Framework\TestCase;

class PluginInfoTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $sut = new PluginInfo(
            "./plugins/feedview/",
            new FakeSystemChecker,
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["feedview"])
        );
        $response = $sut();
        $this->assertEquals("Feedview 1.1-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
