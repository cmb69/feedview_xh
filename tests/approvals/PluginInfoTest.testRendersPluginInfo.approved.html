<!-- feedview info -->
<h1>Feedview 1.3-dev</h1>
<div>
  <h2>System check</h2>
  <p class="xh_fail">PHP version is at least 7.2.0: failure</li>
  <p class="xh_fail">PHP extension 'libxml' is loaded: failure</li>
  <p class="xh_fail">CMSimple_XH version is at least 1.7.0: failure</li>
  <p class="xh_warning">'./plugins/feedview/config/' is writable: warning</li>
  <p class="xh_warning">'./plugins/feedview/css/' is writable: warning</li>
  <p class="xh_warning">'./plugins/feedview/languages/' is writable: warning</li>
</div>
